pkg/importer/dataStream.go

```diff
@@ -734,31 +920,51 @@ func (d *DataStream) convertQcow2ToRaw(src, dest string) error {
 }
 
 // Copy endpoint to dest based on passed-in reader.
-func (d *DataStream) copy(dest string) error {
+func (d *DataStream) copy(dest string) (err error) {
 	if d.isHTTPQcow2() {
-		err := d.convertQcow2ToRawStream(dest)
+		err = d.convertQcow2ToRawStream(dest)
 		if err != nil {
 			return err
 		}
 	} else {
 		dest = filepath.Clean(dest)
 		glog.V(2).Infof("copying image file to %q", dest)
 		tmpDest := dest
-		if d.qemu {
-			// copy to tmp; qemu conversion will write to passed-in destination
-			tmpDest = randTmpName(dest)
-			glog.V(3).Infof("Copy: temp file for qcow2 conversion: %q", tmpDest)
-			defer func(f string) {
-				os.Remove(f)
-			}(tmpDest)
+		if d.isCopyToTmpRequired() {
+			if d.isQemu() {
+				// copy to tmp; qemu conversion will write to passed-in destination
+				tmpDest = randTmpName(dest)
+				glog.V(3).Infof("Copy: temp file for qcow2 conversion: %q", tmpDest)
+				defer func(f string) {
+					os.Remove(f)
+				}(tmpDest)
+			}
+			if d.isMovble() {
+				//before moving has to close all readers
+				closeReaders(d.Readers)
```

Comment:
Shouldn't be necessary

Reply:
At this stage there are Readers on the file,either Rename will fail or the file handle sill become stale and Close in dataStream.Close() will throw an error

Reply:
That is very surprising to me.  Renaming an open file is totally fine at the POSIX level so I wonder where/how this error is triggered.

Reply:
You are absolutely right!
 i got confused with src opened file vs dest opened file. Pushing a fix for that